maint/gen/codegen.c

```diff
@@ -741,22 +741,24 @@ generate_decoder(FILE *out, struct syscall *syscall, bool is_variant,
 	}
 
 	if (out_ptrs == 0) {
-		if (syscall->is_ioctl) {
-			OUTFI("tprint_arg_next();\n");
-		}
-
 		/* 0 out ptrs: print all args in sysenter */
 		for (size_t i = arg_offset; i < syscall->arg_count; i++) {
 			struct syscall_argument arg = syscall->args[i];
 			OUTFI("/* arg: %s (%s) */\n",
 			      arg.name, type_to_ctype(arg.type));
+			if (i == arg_offset) {
+				if (syscall->is_ioctl)
+					OUTFI("tprints_arg_next_name(\"%s\");\n", arg.name);
+				else
+					OUTFI("tprints_arg_name(\"%s\");\n", arg.name);
+			}
 			CLEANUP_FREE char *arg_val =
 				get_syscall_arg_value(syscall, arg_index++);
 			generate_printer(out, syscall, arg.name, arg_val, true,
 					 arg.type, indent_level);
 
 			if (i < syscall->arg_count - 1) {
-				OUTSI("tprint_arg_next();\n");
+				OUTFI("tprints_arg_next_name(\"%s\");\n", arg.name);
```

Comment:
`arg.name` here is the name of the argument that's already printed.
I suggest removing this `if` statement altogether.